fdbserver/VersionedBTree.actor.cpp

```diff
@@ -2871,50 +2942,60 @@ class DWALPager final : public IPager2 {
 	                                                           bool header) {
 		ASSERT(!self->memoryOnly);
 
-		// if (g_network->getCurrentTask() > TaskPriority::DiskRead) {
-		// 	wait(delay(0, TaskPriority::DiskRead));
-		// }
-
 		state Reference<ArenaPage> page =
-		    header ? Reference<ArenaPage>(new ArenaPage(smallestPhysicalBlock, smallestPhysicalBlock))
-		           : self->newPageBuffer();
-		debug_printf("DWALPager(%s) op=readPhysicalStart %s ptr=%p\n",
+		    header ? ArenaPage::construct(smallestPhysicalBlock, smallestPhysicalBlock) : self->newPageBuffer();
+		debug_printf("DWALPager(%s) op=readPhysicalStart %s ptr=%p header=%d\n",
 		             self->filename.c_str(),
 		             toString(pageID).c_str(),
-		             page->begin());
-
-		state PriorityMultiLock::Lock lock = wait(self->ioLock.lock(std::min(priority, ioMaxPriority)));
```

Comment:
Was the priority lock intentionally removed here?

Reply:
Yes, ReadPhysicalPage and ReadPhysicalMultiPage both use 1 or more concurrent calls to ReadPhysicalBlock which is where the IO lock is taken.